File: frontend/src/types/feedback.ts
export interface FeedbackCreate {
  event_id: number;
  attendee_id: number;
  overall_rating: number;
  venue_rating?: number;
  organization_rating?: number;
  speaker_rating?: number;
  catering_rating?: number;
  comments?: string;
}

export interface FeedbackSummary {
  event_id: number;
  total_submissions: number;
  avg_overall: number;
  avg_venue: number;
  avg_organization: number;
  avg_speaker: number;
  avg_catering: number;
}

File: backend/app/schemas/feedback.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

class FeedbackCreate(BaseModel):
    event_id: int
    attendee_id: int
    overall_rating: int = Field(ge=1, le=5)
    venue_rating: Optional[int] = Field(default=None, ge=1, le=5)
    organization_rating: Optional[int] = Field(default=None, ge=1, le=5)
    speaker_rating: Optional[int] = Field(default=None, ge=1, le=5)
    catering_rating: Optional[int] = Field(default=None, ge=1, le=5)
    comments: Optional[str] = None

class FeedbackResponse(BaseModel):
    id: int
    event_id: int
    attendee_id: int
    overall_rating: int
    venue_rating: Optional[int] = None
    organization_rating: Optional[int] = None
    speaker_rating: Optional[int] = None
    catering_rating: Optional[int] = None
    comments: Optional[str] = None
    submitted_at: datetime

    class Config:
        from_attributes = True

class FeedbackSummary(BaseModel):
    event_id: int
    total_submissions: int
    avg_overall: float
    avg_venue: float
    avg_organization: float
    avg_speaker: float
    avg_catering: float